vital/tests/test_image_container_set.cxx

```diff
@@ -0,0 +1,409 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <gtest/gtest.h>
+
+#include <vital/types/image.h>
+#include <vital/types/image_container_set_simple.h>
+
+
+namespace {
+
+auto test_logger = kwiver::vital::get_logger( "vital.tests.test_image_container_set" );
+
+}
+
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char* argv[] )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+// Helper functions / classes for tests
+namespace {
+
+using ic_vec_t = std::vector< kwiver::vital::image_container_sptr >;
+
+/**
+ * Create simple vector of image_container_sptr instances.
+ *
+ * Output will consist of 3 image container sptrs of slightly increasing size:
+ * - 1x1
+ * - 2x2
+ * - 3x3
+ */
+ic_vec_t make_simple_ic_vec()
+{
+  using namespace kwiver::vital;
+
+  using img_vec_t = std::vector< image_container_sptr >;
+  img_vec_t img_vec;
+  img_vec.push_back( std::make_shared< simple_image_container >( image( 1, 1 ) ) );
+  img_vec.push_back( std::make_shared< simple_image_container >( image( 2, 2 ) ) );
+  img_vec.push_back( std::make_shared< simple_image_container >( image( 3, 3 ) ) );
+  return img_vec;
+}
+
+} // end namespace: anonymous
+
+// ----------------------------------------------------------------------------
+TEST( simple_image_container_set, empty )
+{
+  kwiver::vital::image_container_set_sptr empty_set;
+  empty_set = std::make_shared<kwiver::vital::simple_image_container_set>();
+
+  // Check size is reported as zero
+  EXPECT_EQ(0, empty_set->size()) << "Set empty";
+}
+
+// ----------------------------------------------------------------------------
+// Test construction with a non-empty vector of image_container_sptrs
+TEST( simple_image_container_set, construct_nonempty )
+{
+  using namespace kwiver::vital;
+  // Make a vector of empty images.
+  ic_vec_t img_vec = make_simple_ic_vec();
+  simple_image_container_set sics( img_vec );
+}
+
+// ----------------------------------------------------------------------------
+// Test size return from default construction
+TEST( simple_image_container_set, size_empty )
+{
+  using namespace kwiver::vital;
+  simple_image_container_set sics;
+  EXPECT_EQ( sics.size(), 0 );
+}
+
+// ----------------------------------------------------------------------------
+// Test set non-const iteration
+TEST( simple_image_container_set, expected_iteration )
+{
+  using namespace kwiver::vital;
+  // Make a vector of empty images.
+  ic_vec_t img_vec = make_simple_ic_vec();
+  simple_image_container_set sics( img_vec );
+
+  // We should be able to iterate 3 times with img_vec and the returns should
+  // be equivalent to the direct vector iterator
+  ic_vec_t::iterator                   vec_it = img_vec.begin();
+  simple_image_container_set::iterator sic_it = sics.begin();
```

Comment:
Curious why we're using being() to start and cend() below. Shouldn't those be consistent depending on whether the iterator is const? Probably it doesn't matter in this context.